### PR TITLE
feat: add sum_of_binomial_coefficients

### PR DESCRIPTION
**Pull request by muskan0719**
#### Description of Change
An Algorithm to find the sum of binomial coefficients for a given positive integer.
The sum of binomial coefficients for a given n is calculated by the formula 2^n. 
Example 1-
n=2 
sum=2^2=4

Example 2-
n=4
sum=2^4=16

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1363"></a>

### CODE REVIEW

File: math/sum_of_binomial_coefficient.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * @file
+ * @brief Algorithm to find sum of binomial coefficients of a given positive integer.
+ * @details Given a positive integer n, the task is to find the sum of binomial coefficient i.e
+ * nC0 + nC1 + nC2 + ……. + nCn-1 + nCn
+ * By induction, we can prove that the sum is equal to 2^n
+ * @source https://en.wikipedia.org/wiki/Binomial_coefficient#Sums_of_the_binomial_coefficients
+ * @author muskan0719 [https://github.com/muskan0719]
+ */
+#include<iostream> /// for std::cin and std::cout
+#include<cassert>  /// for assert
```
Comment (Panquesito7) [inline math/sum_of_binomial_coefficient.cpp:11]:
```suggestion
#include <iostream> /// for std::cin and std::cout
#include <cassert>  /// for assert

```

File: math/sum_of_binomial_coefficient.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ } 
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() 
+{ 
+    test(); // execute the tests
+	return 0;
```
Comment (Panquesito7) [inline math/sum_of_binomial_coefficient.cpp:59]:
```suggestion
    return 0;
```

File: math/sum_of_binomial_coefficient.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ * @details Given a positive integer n, the task is to find the sum of binomial coefficient i.e
+ * nC0 + nC1 + nC2 + ……. + nCn-1 + nCn
+ * By induction, we can prove that the sum is equal to 2^n
+ * @source https://en.wikipedia.org/wiki/Binomial_coefficient#Sums_of_the_binomial_coefficients
+ * @author muskan0719 [https://github.com/muskan0719]
+ */
+#include<iostream> /// for std::cin and std::cout
+#include<cassert>  /// for assert
+
+/**
```
Comment (Panquesito7) [inline math/sum_of_binomial_coefficient.cpp:13]:
```suggestion
/**
 * @namespace math
 * @brief Mathematical algorithms
 */
namespace math {
/** 
```

File: math/sum_of_binomial_coefficient.cpp [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+/** 
+ * Function to calculate sum of binomial coefficients
+ * @param n number
+ * @return Sum of binomial coefficients of number 
+ */
+int binomialCoeffSum(int n) 
+{ 
+    //Calculating 2^n  
+	return (1 << n); 
+}
```
Comment (Panquesito7) [inline math/sum_of_binomial_coefficient.cpp:22]:
```suggestion
}
}  // namespace math
```

File: math/sum_of_binomial_coefficient.cpp [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+/** 
+ * Function to calculate sum of binomial coefficients
+ * @param n number
+ * @return Sum of binomial coefficients of number 
+ */
+int binomialCoeffSum(int n) 
+{ 
+    //Calculating 2^n  
+	return (1 << n);
```
Comment (Panquesito7) [inline math/sum_of_binomial_coefficient.cpp:21]:
```suggestion
    // Calculating 2^n  
    return (1 << n); 
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/muskan0719/C-Plus-Plus/actions

File: math/sum_of_binomial_coefficient.cpp [hunk 6]
```diff
@@ -0,0 +1,59 @@
+/**
+ * @file
+ * @brief Algorithm to find sum of binomial coefficients of a given positive
+ * integer.
+ * @details Given a positive integer n, the task is to find the sum of binomial
+ * coefficient i.e nC0 + nC1 + nC2 + ……. + nCn-1 + nCn By induction, we can
+ * prove that the sum is equal to 2^n
+ * @source
+ * https://en.wikipedia.org/wiki/Binomial_coefficient#Sums_of_the_binomial_coefficients
+ * @author muskan0719 [https://github.com/muskan0719]
```
Comment (Panquesito7) [inline math/sum_of_binomial_coefficient.cpp:10]:
```suggestion
 * @author [muskan0719](https://github.com/muskan0719)
```

File: math/sum_of_binomial_coefficient.cpp [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+ * coefficient i.e nC0 + nC1 + nC2 + ……. + nCn-1 + nCn By induction, we can
+ * prove that the sum is equal to 2^n
+ * @source
+ * https://en.wikipedia.org/wiki/Binomial_coefficient#Sums_of_the_binomial_coefficients
+ * @author muskan0719 [https://github.com/muskan0719]
+ */
+#include <cassert>   /// for assert
+#include <iostream>  /// for std::cin and std::cout
+
+/**
```
Comment (Panquesito7) [inline math/sum_of_binomial_coefficient.cpp:15]:
```suggestion
/**
 * @namespace math
 * @brief Mathematical algorithms
 */
namespace math {
/**
```

File: math/sum_of_binomial_coefficient.cpp [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+}
+}  // namespace math
+
+/**
+ * Function for testing binomialCoeffSum function.
+ * test cases and assert statement.
+ * @returns `void`
+ */
+static void test() {
+    int test_case_1 = binomialCoeffSum(2);
```
Comment (Panquesito7) [inline math/sum_of_binomial_coefficient.cpp:32]:
```suggestion
    int test_case_1 = math::binomialCoeffSum(2);

```

File: math/sum_of_binomial_coefficient.cpp [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    assert(test_case_1 == 4);
+    std::cout << "Test_case_1 Passed!" << std::endl;
+
+    int test_case_2 = binomialCoeffSum(3);
```
Comment (Panquesito7) [inline math/sum_of_binomial_coefficient.cpp:36]:
```suggestion
    int test_case_2 =  math::binomialCoeffSum(3);

```

File: math/sum_of_binomial_coefficient.cpp [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    assert(test_case_2 == 8);
+    std::cout << "Test_case_2 Passed!" << std::endl;
+
+    int test_case_3 = binomialCoeffSum(4);
```
Comment (Panquesito7) [inline math/sum_of_binomial_coefficient.cpp:40]:
```suggestion
    int test_case_3 =  math::binomialCoeffSum(4);

```

File: math/sum_of_binomial_coefficient.cpp [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    assert(test_case_3 == 16);
+    std::cout << "Test_case_3 Passed!" << std::endl;
+
+    int test_case_4 = binomialCoeffSum(5);
```
Comment (Panquesito7) [inline math/sum_of_binomial_coefficient.cpp:44]:
```suggestion
    int test_case_4 =  math::binomialCoeffSum(5);

```

File: math/sum_of_binomial_coefficient.cpp [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+    assert(test_case_4 == 32);
+    std::cout << "Test_case_4 Passed!" << std::endl;
+
+    int test_case_5 = binomialCoeffSum(7);
```
Comment (Panquesito7) [inline math/sum_of_binomial_coefficient.cpp:48]:
```suggestion
    int test_case_5 =  math::binomialCoeffSum(7);

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Review by Panquesito7 — APPROVED** [review]
Nice work; LGTM. 😄 👍

**Comment by muskan0719** [comment]
@kvedala please review this.

File: math/sum_of_binomial_coefficient.cpp [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+ * @brief Mathematical algorithms
+ */
+namespace math {
+
+	/** 
+	 * Function to calculate sum of binomial coefficients
+	 * @param n number
+	 * @return Sum of binomial coefficients of number 
+	 */
+	int binomialCoeffSum(int n)
```
Comment (ayaankhan98) [inline math/sum_of_binomial_coefficient.cpp:24]:
```suggestion
	uint64_t binomialCoeffSum(uint64_t n) 
```
as `n >= 0` always

**Review by ayaankhan98 — APPROVED** [review]
LGTM! Thanks for your contribution @muskan0719

**Pull request merged**